### DISCUSSION TITLE
OpenChrom Batch Process Help

### DISCUSSION THREAD

**Discussion by HeyhoTobi** [OpenChrom/openchrom#547] [Q&A] [answered]
Hallo guys,
I am very new to OpenChrom, but I need a tool to automatically analyze my GC-MS data, because it´s a huge amount.
Therefore, I found this program and created a batch process:
![grafik](https://github.com/user-attachments/assets/e009bfc4-09d7-4626-9568-0ffe285dee9a)

But the problem is that if I put too many samples to process the memory of 2GB gets filled up very fast - so how to change that? tried with a openchrom.ini file, but didn´t work.
And the more important thing is that after the batch process I only get a .csv file with only the titles but no data (especially area for specific substances).

Would be very nice if you could help me, thanks a lot!
Best, Tobi

**Comment by eselmeister** [comment, marked as answer]
Basically, it's the correct choice to increase the available RAM for OpenChrom to better handle large files. Did you switch the `-Xmx` setting from default 2 GB:

`-Xmx2g`

to e.g. 6 GB:

`-Xmx6g`

OpenChrom must be shutdown before changing the setting.

**Reply by HeyhoTobi** [reply]
Ok, now it´s working, just changed again the settings. Thanks!

and regarding my batch process: is there any better way to process my files automatically? 
I have in total 700 chromatograms to analyze..

I now changed the Peak Detector to [Template] (no UI) and finally get values in the .csv files.

**Reply by eselmeister** [reply]
It depends on the goals you'd like to achieve. I recommend to first edit a few chromatograms manually. You can use the macro recorder to create a process method. When you're happy with the results, use the process method in your batch.

![image](https://github.com/user-attachments/assets/7b789990-e708-41f2-88b4-c0a5d7efad23)

**Reply by HeyhoTobi** [reply]
Already played around manually and found my way to go to automate it.

Well I focus mostly on aldehydes and ketones in food matrices and use 6 standards + found some other substances of interest. so I created a template for the peak detection with retention times and same for integration.
So in the end mostly interested in Substance + Area.

**Reply by eselmeister** [reply]
Do you also want to apply an external calibration?

**Reply by HeyhoTobi** [reply]
I plan to process the .csv files from openchrom (substances + areas) with a python script for quantification, etc.

**Reply by eselmeister** [reply]
Have a look, we also offer a quantitation add-on:
https://lablicate.com/marketplace/extensions/quantitation

It's free for students/PhDs. An it can be incorporated in the batch process. Let me know if I shall help you to set it up.

**Reply by HeyhoTobi** [reply]
Ok, thanks a lot! I will definitely give it a try! ;)
